x-pack/plugins/security_solution/server/endpoint/ingest_integration.ts

```diff
@@ -33,54 +34,62 @@ export const getPackageConfigCreateCallback = (
     // with diffs from last dispatched manifest, if it exists
     const snapshot = await manifestManager.getSnapshot({ initialize: true });
 
-    if (snapshot === null) {
-      logger.warn('No manifest snapshot available.');
-      return updatedPackageConfig;
-    }
-
-    if (snapshot.diffs.length > 0) {
-      // create new artifacts
-      await manifestManager.syncArtifacts(snapshot, 'add');
+    let success = true;
+    try {
+      if (snapshot && snapshot.diffs.length > 0) {
+        // create new artifacts
+        const errors = await manifestManager.syncArtifacts(snapshot, 'add');
```

Comment:
how does syncArtifacts relates getPackageConfigCreateCallback. That appears to update something, perhaps a side effect?

Reply:
@nnamdifrankie If new exception list entries were created, we may need to create new artifacts before sending the updated manifest. That's what `syncArtifacts` does.

Reply:
@XavierM and I have been talking about how to improve the way this callback happens... I think we will need some changes to make this more robust. Happy to discuss it @nnamdifrankie

Reply:
I think a test around what branches are supposed to exist together will suffice for now. Not sure what is the relationship between the snapshot and new package config.

Reply:
@nnamdifrankie Just pushed some of the tests, working on more.

Reply:
@nnamdifrankie I've added several tests and quite a bit of manual testing. I think this fix will ensure that policy creation is not interrupted by any of the manifest processing code.